h2o-algos/src/main/java/hex/generic/GenericModel.java

```diff
@@ -131,6 +154,34 @@ protected PredictScoreResult predictScoreImpl(Frame fr, Frame adaptFrm, String d
             return predictScoreMojoImpl(fr, destination_key, j, computeMetrics);
         } else
             return super.predictScoreImpl(fr, adaptFrm, destination_key, j, computeMetrics, customMetricFunc);
+//            return super.predictScoreImpl(fr, adaptFrm, destination_key, j, true, customMetricFunc);
+    }
+
+    private Iced getParamByName(String name) {
+        return Arrays.stream(this._parms._modelParameters)
+                .filter(p -> Objects.equals(p.name, name)).findAny().get().actual_value;
+    }
+
+    @Override
+    public double aic(double likelihood) {
+        // calculate negative loglikelihood specifically  for GLM
+        if (!_algoName.equals("glm")) {
+            return 0;
+        } else {
+            long betasCount = Arrays.stream(((GlmMojoModelBase) this.genModel()).getBeta()).filter(b -> b != 0).count();
+            return -2 * likelihood + 2 * betasCount;
+        }
+    }
+
+    @Override
+    public double likelihood(double w, double y, double[] f) {
```

Comment:
If it's negative log likelihood, I'd prefer to name it `negative_log_likelihood` or something like  that.

Reply:
This is a confusing issue.  We have been using likelihood to mean negative_log_likelihood.  There will be so many changes.  For now I suggest just add a comment indicating that this is actually negative log likelihood function.  Customer is waiting for this one.  Moving forward, I think I want to change it to negative_log_likelihood function.  Here is the issue: https://github.com/h2oai/h2o-3/issues/16059